doc/user_guide/user_guide.md

```diff
@@ -174,3 +225,61 @@ While the `cp` command can rename the copied file in the target location you can
 bfsc cp foo.jar bfs:/drivers/
 bfsc cp bfs:/drivers/foo.jar .
 ```
+
+Command `cp` supports recursive copying:
+
+```shell
+bfsc cp -r Dir bfs:/
+```
+
+However there are some special cases explained in the following sections.
+
+##### Ambigue Entries in BucketFS
+
+BucketFS differes from the local file system as it may contain *ambigue* entries. An ambigue entry represents a regular file and a directory at the same time. Obviously you cannot download both flavors of such an entry to the local file system and BFSC will abort with an error message.
+
+You can however tell BFSC to download *one* of the flavors of such an ambigue entry.
+* For recursively downloading the directory please specify option `-r` *and* append a slash `/` to the name of the ambigue entry.
+* For downloading the regular file omit both of them.
+
+| Command line               | Semantics                                |
+|----------------------------|------------------------------------------|
+| `bfsc cp bfs:/ambigue/`    | error message                            |
+| `bfsc cp -r bfs:/ambigue`  | error message                            |
+| `bfsc cp bfs:/ambigue`     | download regular file `ambigue`          |
+| `bfsc cp -r bfs:/ambigue/` | recursively download directory `ambigue` |
+
+For non-ambigue entries the trailing slash is not relevant.
+
+For ambigue entries on lower levels during recursive download BFSC will report a warning and only download the directory flavor of the ambigue entry.
+
+##### Existing Entries in the Local File System
+
+When downloading from BucketFS the target in the local file system might already exist and could be either a directory or a regular file.
+
+The following table shows all cases assuming the local file system contains a regular file `b.txt` and a folder `B`.
+
+| Command                | Semantics / error message                   |
+|------------------------|---------------------------------------------|
+| `cp bfs://a.txt b.txt` | download regular file `a.txt` and rename it to `b.txt`    |
+| `cp bfs://a.txt B`     | download regular file `a.txt` into existing local directory `B` and keep the original filename resulting in `B/a.txt`  |
+| `cp -r bfs://A B`      | recursively download directory `A` to `B/A` |
+| `cp -r bfs://A b.txt`  | error message                               |
+
+When downloading a directory from BucketFS you can rename the directory by specifying a local name that does not exist, yet.
+
+#### Deleting Files from a Bucket
+
+With `bfsc rm` you can delete files in BucketFS.
+
+Examples
+```shell
+bfsc rm a.txt
+bfsc rm folder/b.txt
+bfsc rm -r folder
```

Comment:
```suggestion
bfsc rm bfs:/a.txt
bfsc rm bfs:/folder/b.txt
bfsc rm -r bfs:/folder
```

Reply:
As bfsc is not designed to remove files in the local filesystem, anyway, I left the protocol prefix intentionally.
But I will add a dedicated test to ensure this works as expected, though.